legacy/builder/utils/utils.go

```diff
@@ -184,29 +184,30 @@ const (
 )
 
 func ExecCommand(ctx *types.Context, command *exec.Cmd, stdout int, stderr int) ([]byte, []byte, error) {
```

Comment:
I got inspired by this refactoring, but do what you want with it. Since it took longer than expected it might not be that good
```
func getWriter(option int, isVerbose bool, candidate io.Writer) io.Writer {
	getCandidateOrStdout := func() io.Writer {
		if candidate != nil {
			return candidate
		}
		return os.Stdout
	}
	mapOptionToWriterFunc := map[int]func() io.Writer{
		Ignore: func() io.Writer {
			return io.Discard
		},
		Show: getCandidateOrStdout,
		ShowIfVerbose: func() io.Writer {
			if !isVerbose {
				return io.Discard
			}
			return getCandidateOrStdout()
		},
		Capture: func() io.Writer {
			return &bytes.Buffer{}
		},
	}
	return mapOptionToWriterFunc[option]()
}

func ExecCommand(ctx *types.Context, command *exec.Cmd, stdout int, stderr int) ([]byte, []byte, error) {
	stdoutWriter := getWriter(stdout, ctx.Verbose, ctx.Stdout)
	stderrWriter := getWriter(stderr, ctx.Verbose, ctx.Stderr)
	return ExecCommandWithWriters(ctx, command, stdoutWriter, stderrWriter)
}

func ExecCommandWithWriters(ctx *types.Context, command *exec.Cmd, stdout io.Writer, stderr io.Writer) ([]byte, []byte, error) {
	if ctx.Verbose {
		ctx.Info(PrintableCommand(command.Args))
	}
	command.Stdout = stdout
	command.Stderr = stderr

	err := command.Start()
	if err != nil {
		return nil, nil, errors.WithStack(err)
	}

	err = command.Wait()

	var outbytes, errbytes []byte
	if buf, ok := command.Stdout.(*bytes.Buffer); ok {
		outbytes = buf.Bytes()
	}
	if buf, ok := command.Stderr.(*bytes.Buffer); ok {
		errbytes = buf.Bytes()
	}

	return outbytes, errbytes, errors.WithStack(err)
}```

Reply:
This is something to keep in mind when this subroutine will be merged with `executils` and removed from `legacy`.